### DISCUSSION TITLE
How to evaluate a switch state that has not changed since boot?

### DISCUSSION THREAD

**Discussion by TPPTim** [arendst/Tasmota#22343] [Q&A] [answered]
### PROBLEM DESCRIPTION
Hello together,
I did not find an solution since the last three hours. So I stated a new issue for that question.

I am trying to evaluate two DIO pin values that I have set up as switches. These two signals come from a charge controller, which stands for CHARGE (switch2) and FINISHED (switch3).
I have connected a digital LED to the ESP. What I want is to change the color of the LED depending on the pin state of the charge controller.

This works when the ESP is booted and the pins changes there state. So I can catch this with rules like 
ON Switch2#state=0 DO Color1 #ABABAB ENDON

The problem is that the rule will not be triggered if the pin state is set before the ESP is fully booted. So there is no rising edge and no pin change event to trigger the rule.

How can I change the LED color after the ESP has fully booted when one of the pins (switches) has HIGH (ON) states?

### REQUESTED INFORMATION

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): ESP8266
- [x] Tasmota binary firmware version number used: [Tasmota 14.3.0 (release-tasmota) by Theo Arends](https://bit.ly/tasmota)
  - [x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
12:52:06.235 RSL: RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
12:52:06.458 RSL: RESULT = {"Module":{"0":"Generic"}}
12:52:06.692 RSL: RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"1376":"WS2812"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"162":"Switch3"},"GPIO13":{"0":"None"},"GPIO14":{"161":"Switch2"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"10624":"ADC Voltage"}}

```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
12:54:11.873 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":87,"Free":424,"Rules":"ON Switch2#state=1 DO Color1 #00807A ENDON  ON Switch3#state=1 DO Color1 #008000 ENDON "}}
12:54:12.105 RSL: RESULT = {"Rule2":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":127,"Free":384,"Rules":"ON Power1#Boot DO Color1 #C900AE ENDON  ON Switch3#state=0 DO Color1 #C900AE ENDON  ON Switch2#state=0 DO Color1 #C900AE ENDON "}}
12:54:12.307 RSL: RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
12:54:27.882 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_4CB5B8","ButtonTopic":"0","Power":"1","PowerLock":"0","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,2,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
12:54:27.933 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"External System","Uptime":"0T00:03:06","StartupUTC":"2024-10-22T11:51:21","Sleep":50,"CfgHolder":4617,"BootCount":42,"BCResetTime":"2024-10-18T13:16:19","SaveCount":298,"SaveAddress":"FA000"}}
12:54:27.978 RSL: STATUS2 = {"StatusFWR":{"Version":"14.3.0(release-tasmota)","BuildDateTime":"2024-10-15T08:18:01","Boot":31,"Core":"2_7_7","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"437/699"}}
12:54:27.007 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Classic Games Society","Konfiguration"],"TelePeriod":300,"Resolution":"558580C0","SetOption":["0000A009","2805C80001000600003C5A0A192800000000","00800080","00006000","00004001","00000000"]}}
12:54:28.025 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":649,"Free":352,"Heap":25,"ProgramFlashSize":1024,"FlashSize":4096,"FlashChipId":"164068","FlashFrequency":40,"FlashMode":"DOUT","Features":["0809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080","00000000"],"Drivers":"1,2,!3,4,!5,!6,7,!8,9,10,12,!16,!18,!19,!20,!21,!22,!24,26,!27,29,!30,!35,!37,!45,62,!68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
12:54:28.079 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-4CB5B8-5560","IPAddress":"192.168.188.29","Gateway":"192.168.188.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.188.1","DNSServer2":"0.0.0.0","Mac":"48:3F:DA:4C:B5:B8","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
12:54:28.116 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_4CB5B8","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
12:54:28.146 RSL: STATUS7 = {"StatusTIM":{"UTC":"2024-10-22T11:54:28Z","Local":"2024-10-22T12:54:28","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+01:00","Sunrise":"07:22","Sunset":"17:46"}}
12:54:28.215 RSL: STATUS10 = {"StatusSNS":{"Time":"2024-10-22T12:54:28","Switch2":"OFF","Switch3":"OFF","ANALOG":{"Voltage":4.14}}}
12:54:28.237 RSL: STATUS11 = {"StatusSTS":{"Time":"2024-10-22T12:54:28","Uptime":"0T00:03:07","UptimeSec":187,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":21,"MqttCount":0,"POWER":"ON","Dimmer":79,"Color":"C900AE","HSBColor":"308,100,79","Channel":[79,0,68],"Scheme":0,"Width":1,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"Classic Games Society","BSSId":"E0:28:6D:99:D5:13","Channel":11,"Mode":"11n","RSSI":100,"Signal":-35,"LinkCount":1,"Downtime":"0T00:00:04"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
Don't start issues to ask questions, that's what Discussions (where your thread was moved), or the Discord server, is for.

You should be able to have boot-time rules to act on initial Switch state using triggers like `Switch1#Boot`

**Comment by TPPTim** [comment]
Thank you for the hint. I will know it for the future :)
Its working well with `Switch1#Boot`.
